cpu/sam0_common/periph/rtc.c

```diff
@@ -208,14 +265,30 @@ void rtc_clear_alarm(void)
 
 void rtc_poweron(void)
 {
+#ifdef REG_RTC_MODE2_CTRLA
     RTC->MODE2.CTRLA.bit.ENABLE = 1;
-    while (RTC->MODE2.SYNCBUSY.bit.ENABLE);
+#else
+    RTC->MODE2.CTRL.bit.ENABLE = 1;
+#endif
+#ifdef REG_RTC_MODE2_SYNCBUSY
+    while (RTC->MODE2.SYNCBUSY.bit.ENABLE) {}
+#else
+    while (RTC->MODE2.STATUS.bit.SYNCBUSY) {}
+#endif
 }
 
 void rtc_poweroff(void)
 {
+#ifdef REG_RTC_MODE2_CTRLA
     RTC->MODE2.CTRLA.bit.ENABLE = 0;
-    while (RTC->MODE2.SYNCBUSY.bit.ENABLE);
+#else
+    RTC->MODE2.CTRL.bit.ENABLE = 0;
+#endif
+#ifdef REG_RTC_MODE2_SYNCBUSY
```

Comment:
I would replace these `#ifdef REG_RTC_MODE2_SYNCBUSY` block by a static or static inline function here. Easier to read and more compact.

Reply:
like so?